File: rnd/blog/logging.py
import logging
import os
import sys


logger = logging.getLogger('blog')


def enable_logger(verbose=False):
    global logger

    if verbose:
        log_level = logging.getLevelName('DEBUG')
    else:
        log_level = logging.getLevelName('INFO')

    logger.setLevel(log_level)
    logger.handlers = []
    handler = logging.StreamHandler(sys.stderr)
    handler.setFormatter(logging.Formatter('%(levelname)s: %(message)s'))
    logger.addHandler(handler)

File: _plugins/blog/logging.rb
# frozen_string_literal: true

require 'jekyll'

module Blog
  # Logging
  module Logging
    def info(msg)
      Jekyll.logger.info 'blog:', msg
    end

    def debug(msg)
      Jekyll.logger.debug 'blog:', msg
    end

    def error(msg)
      Jekyll.logger.error 'blog:', msg
    end

    def warn(msg)
      Jekyll.logger.warn 'blog:', msg
    end
  end
end
